cpp/src/arrow/dataset/CMakeLists.txt

```diff
@@ -20,6 +20,11 @@ add_custom_target(arrow_dataset)
 # Headers: top level
 arrow_install_all_headers("arrow/dataset")
 
+# pkg-config support
+configure_file(arrow-dataset.pc.in "${CMAKE_CURRENT_BINARY_DIR}/arrow-dataset.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/arrow-dataset.pc"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
```

Comment:
Could you use `arrow_add_pkg_config()`?

Reply:
Thanks! I didn't know the function was defined.
I've fixed it.